src/api/c/libspeechd.c

```diff
@@ -517,6 +529,7 @@ SPDConnection *spd_open2(const char *client_name, const char *connection_name,
 				assert(0);
 			SPD_DBG(*error_result);
 			close(connection->socket);
+			free(connection->buf);
```

Comment:
? No, buf is not allocated at this point, and the pointer value is even undefined